### PR TITLE
Ask for confirmation before deleting nodes and subtrees.

### PR DESCRIPTION
**Pull request by AAJELLAL**

### CODE REVIEW

File: src/components/graph/menus/create-node-menu.js [hunk 1]
```diff
@@ -14,6 +14,43 @@ import { useSelector } from 'react-redux';
...
 import ChildMenuItem from './create-child-menu-item';
+import { CustomDialog } from '../../utils/custom-dialog';
+
+export const NodeActions = {
+    REMOVE_NODE: 'REMOVE_NODE',
+    REMOVE_SUBTREE: 'REMOVE_SUBTREE',
+    NO_ACTION: 'NO_ACTION',
+};
+
+export const getNodeChildren = (treeModel, sourceNodeIds, allChildren) => {
```
Comment (SlimaneAmar) [inline src/components/graph/menus/create-node-menu.js:25]: All pptional Chainings (var?) are really necessary ?

File: src/components/graph/menus/create-node-menu.js [hunk 2]
```diff
@@ -276,20 +316,63 @@ const CreateNodeMenu = ({
...
+    );
+
+    const handleOnValidate = useCallback(() => {
+        if (nodeAction === NodeActions.REMOVE_NODE) {
+            handleNodeRemoval(activeNode);
+        } else {
+            handleRemovalSubtree(activeNode);
+        }
+        handleClose();
+        setNodeAction(NodeActions.NO_ACTION);
```
Comment (FranckLecuyer) [inline src/components/graph/menus/create-node-menu.js:339]: You can call handeOnClose instead of these 2 successive calls ...

**Review by FranckLecuyer — APPROVED** [review]
Code Review: OK (with a small optional remark)
Tests: OK

File: src/components/graph/menus/create-node-menu.js (hunk 1 above)
Comment (AAJELLAL) [reply src/components/graph/menus/create-node-menu.js:25]: Good catch, not all of them.Done.

File: src/components/graph/menus/create-node-menu.js (hunk 2 above)
Comment (AAJELLAL) [reply src/components/graph/menus/create-node-menu.js]: Indeed

**Review by ayolab — APPROVED** [review]

**Pull request merged**